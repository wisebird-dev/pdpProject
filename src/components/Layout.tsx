import React from 'react';

export interface LayoutProps {
    children: React.ReactChildren;
}

const Layout: React.FC<LayoutProps> = ({ children }: LayoutProps) => (
    <>
        <h1>nav bar element</h1>
        {children}
        <footer>footer element</footer>
    </>
);

export default Layout;
