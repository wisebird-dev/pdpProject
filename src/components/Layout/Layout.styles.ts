import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(({ spacing }) => ({
    layout: {
        display: 'grid',
        placeContent: 'center',
        minHeight: spacing(50),
        marginTop: spacing(6),
        marginBottom: spacing(20),
    },
}));

export default useStyles;
