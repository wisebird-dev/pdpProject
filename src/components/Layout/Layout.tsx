import React from 'react';
import Container from '@material-ui/core/Container';

import { Typography } from 'components';

export interface LayoutProps {
    children: React.ReactElement;
}

const Layout: React.FC<LayoutProps> = ({ children }: LayoutProps) => (
    <>
        <Typography variant="h1">nav bar element</Typography>
        <Container>
            <main>{children}</main>
        </Container>
        <footer>
            <Typography variant="body2">footer element</Typography>
        </footer>
    </>
);

export default Layout;
