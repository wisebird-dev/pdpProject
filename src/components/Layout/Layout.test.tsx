import React from 'react';

import { render } from '@testUtils';

import Layout from './Layout';

describe('Layout suite', () => {
    it('renders correctly', () => {
        const {
            container: { firstChild },
        } = render(<Layout>mockChildren</Layout>);

        expect(firstChild).toMatchSnapshot();
    });
});
