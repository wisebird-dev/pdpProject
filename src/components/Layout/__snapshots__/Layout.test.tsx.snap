// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout suite renders correctly 1`] = `
<div
  class="MuiBox-root MuiBox-root-2"
>
  <div
    class="makeStyles-navBar-3"
  >
    <div>
      <button
        aria-label="toggle menu"
        class="MuiButtonBase-root MuiIconButton-root"
        data-testid="toggle-sidemenu-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="PrivateSwipeArea-root-22 PrivateSwipeArea-anchorLeft-23"
        style="width: 20px;"
      />
    </div>
    <h1
      class="MuiTypography-root makeStyles-siteNameMobile-9 MuiTypography-h5 MuiTypography-colorTextSecondary"
    >
      P
    </h1>
    <h1
      class="MuiTypography-root makeStyles-siteName-8 MuiTypography-h5 MuiTypography-colorTextSecondary"
    >
      PDP project
    </h1>
    <div
      aria-expanded="false"
      class="makeStyles-search-10"
      data-testid="autocomplete"
      role="combobox"
    >
      <div
        class="makeStyles-searchIcon-11"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
      <div
        aria-autocomplete="list"
        aria-expanded="true"
        autocapitalize="none"
        class="MuiInputBase-root makeStyles-inputRoot-12 Mui-focused"
        role="combobox"
        spellcheck="false"
      >
        <input
          aria-label="search"
          autocomplete="off"
          class="MuiInputBase-input makeStyles-inputInput-13"
          data-testid="search-input"
          id="search-input"
          placeholder="Search..."
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    class="MuiContainer-root makeStyles-layout-1"
  >
    <div
      class="MuiBox-root MuiBox-root-16"
    >
      <main>
        mockChildren
      </main>
    </div>
  </div>
  <article>
    <footer
      class="makeStyles-footer-17"
    >
      <div
        class="makeStyles-container-18"
      >
        <div
          class="makeStyles-containerItem-19"
        >
          <h5
            class="MuiTypography-root makeStyles-section-20 MuiTypography-body1"
          >
            Social
          </h5>
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
            href="https://www.linkedin.com/in/spyros-goulas"
            rel="noopener"
            target="_blank"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              linkedIn
            </p>
          </a>
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
            href="https://stackoverflow.com/users/6950690/%ce%a3%cf%80%cf%8d%cf%81%ce%bf%cf%82-%ce%93%ce%bf%cf%8d%ce%bb%ce%b1%cf%82"
            rel="noopener"
            target="_blank"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              stack overflow
            </p>
          </a>
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
            href="https://dev.to/sgoulas"
            rel="noopener"
            target="_blank"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              DEV.to
            </p>
          </a>
        </div>
        <div
          class="makeStyles-containerItem-19"
        >
          <h5
            class="MuiTypography-root makeStyles-section-20 MuiTypography-body1"
          >
            FAQ
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            about this project
          </p>
        </div>
        <div
          class="makeStyles-containerItem-19"
        >
          <h5
            class="MuiTypography-root makeStyles-section-20 MuiTypography-body1"
          >
            github
          </h5>
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
            href="https://github.com/sgoulas/pdpProject"
            rel="noopener"
            target="_blank"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              project repository
            </p>
          </a>
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
            href="https://github.com/sgoulas"
            rel="noopener"
            target="_blank"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              github profile
            </p>
          </a>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-middle"
      />
      <p
        class="MuiTypography-root makeStyles-text-21 MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-alignRight"
      >
        made by Spyros Goulas
      </p>
    </footer>
  </article>
</div>
`;
